modules/dhcp_common/server.rb

```diff
@@ -56,10 +56,23 @@ def all_hosts(subnet)
     end
 
     def find_record(subnet_address, an_address)
-      service.find_host_by_ip(subnet_address, an_address) ||
-        service.find_host_by_mac(subnet_address, an_address) ||
-        service.find_lease_by_ip(subnet_address, an_address) ||
-        service.find_lease_by_mac(subnet_address, an_address)
+      records = find_records_by_ip(subnet_address, an_address) ||
+        [find_record_by_mac(subnet_address, an_address)]
+      return if records.empty?
+      records.first
+    end
+
+    def find_record_by_mac(subnet_address, mac_address)
+      service.find_host_by_mac(subnet_address, mac_address) ||
+        service.find_lease_by_mac(subnet_address, mac_address)
+    end
+
+    def find_records_by_ip(subnet_address, ip)
+      hosts = service.find_hosts_by_ip(subnet_address, ip)
+      return hosts if hosts
+      lease = service.find_lease_by_ip(subnet_address, ip)
+      return [lease] if lease
+      nil
```

Comment:
I think this should return an empty list (my mental map is: function returns a single object: an object or nil; function returns a list: a list of of objects or an empty list).